store/store.go

```diff
@@ -221,7 +230,9 @@ func (s *store) Set(nodePath string, dir bool, value string, expireTime time.Tim
 		e.PrevNode = prev.Node
 	}
 
-	s.WatcherHub.notify(e)
+	if expireOpts.ExpireTime.IsZero() || !expireOpts.Refresh || ((getErr != nil && getErr.ErrorCode == etcdErr.EcodeKeyNotFound) || n.Value != value) {
```

Comment:
@cchamplin Have you tried the case you described? I think if we set ttl to 0, the key will still be inserted into etcd and be removed later on. So the watcher should be able to receive the notification anyway.